### ISSUE TITLE
`import.meta.url` error in .mts files

### ISSUE THREAD

**Issue by AlexXanderGrib** [microsoft/TypeScript#49842]
# Bug Report

### 🔎 Search Terms

`import`, `imoprt.meta`, `nodenext`

### 🕗 Version & Regression Information

- This is a crash
- This changed between versions 4.5 and 4.6
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about `ECMA Script modules in Node.JS`
- I was unable to test this on prior versions because there is nothing about it

### ⏯ Playground Link

`4.5.5` - [playground](https://www.typescriptlang.org/play?target=99&moduleResolution=99&module=199&ts=4.5.5&declaration=false&pretty=true#code/JYWwDg9gTgLgdCApjAhnArlANkA)

`4.7.4` (both .ts and .mts filex) - [playground](https://www.typescriptlang.org/play?target=99&moduleResolution=99&module=199&ts=4.7.4&declaration=false&pretty=true#code/JYWwDg9gTgLgdCApjAhnArlANkA)

### 💻 Code

```ts
// .mts file
import.meta.url
```

### 🙁 Actual behavior

```
error TS1470: The 'import.meta' meta-property is not allowed in files which will build into CommonJS output.
```

### 🙂 Expected behavior

It compiles

**Comment by Josh-Cena** [comment]
You did not set `module: NodeNext` in your playground.

[Playground](https://www.typescriptlang.org/play?target=99&moduleResolution=99&module=199&ts=4.7.4&declaration=false&pretty=true#code/JYWwDg9gTgLgdCApjAhnArlANkA)

**Comment by fatcerberus** [comment]
@Josh-Cena `module=199` is in fact part of the playground URL; it looks like the same bug that causes it to select ES3 in the dropdown when it's actually targeting ESNext.  Manually selecting `NodeNext` doesn't fix the error.

**Comment by Josh-Cena** [comment]
Looks like a playground bug. You have to select `NodeNext`, then unselect it, then select it again.

https://user-images.githubusercontent.com/55398995/178129146-1d3f37a4-e4af-4d12-959c-6c7b1a932a77.mov

**Comment by RyanCavanaugh** [comment]
Confirmed this works fine from actual tsc; it does appear to just be the website

**Comment by gajus** [comment]
getting the same error with minimal typescript config:

```json
{
  "compilerOptions": {
    "module": "NodeNext",
    "target": "es5",
    "noEmit": true,
    "types": ["cypress"]
  },
  "extends": "@tsconfig/node16/tsconfig.json",
  "include": ["vite.config.ts", "cypress.config.ts"]
}
```

**Comment by AlanMorel** [comment]
I am seeing this issue with these settings:
```
{
    "compilerOptions": {
        "strict": true,
        "module": "NodeNext",
        "moduleResolution": "Node",
        "jsx": "react-jsx",
        "esModuleInterop": true,
        "types": ["vite/client"],
    },
  }
```
I'm on TypeScript 4.7.4

**Comment by sagargurtu** [comment]
I'm facing the same issue in VSCode with following settings:
```
{
  "compilerOptions": {
    "target": "esnext",
    "lib": ["dom", "esnext"],
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "module": "NodeNext",
    "moduleResolution": "NodeNext"
  }
}
```

Following error messages are displayed by VSCode:
```
// Cannot find module 'pretty-ms' or its corresponding type declarations.
import ms from "pretty-ms";

// The 'import.meta' meta-property is not allowed in files which will build into CommonJS output.ts(1470)
console.log(import.meta.url);
```

TypeScript version: 4.7.4

**Comment by alecmev** [comment]
Same for `"module": "Node16"`, just in case. Looks like `SourceFile.impliedNodeFormat` is wrong, somehow?

https://github.com/microsoft/TypeScript/blob/38e91fb12099b526e9a2cb5ad6aacfc89d5060ca/src/compiler/checker.ts#L32468

**Comment by okbel** [comment]
Can confirm 👍

**Comment by dargmuesli** [comment]
@navya9singh I'm using the just released typescript v4.8.4 and still face this issue. I also don't see how a fix inside the ts website's source should fix an actual typescript issue. I'd vote to reopen this ticket.

**Comment by thyming** [comment]
I'm also running into this issue with typescript 4.9.4. compiler options include:
```json
  "compilerOptions": {
    "strict": true,
    "target": "ES2021",
    "module": "Node16",
```
